File: infrastructures/ddd/csharp/src/Api/Dto/V1/Requests/CreateExampleRequest.cs
using System.ComponentModel.DataAnnotations;
using OddlyDdd.Api.Dto.V1.Infra;

namespace OddlyDdd.Api.Dto.V1.Requests
{
    /// <summary>
    /// Example request DTO for creating an example.
    /// Request DTOs:
    /// - Live in /Api/Dto/
    /// - Have Request suffix
    /// - Used for HTTP transport only
    /// - NO business logic
    /// - Can have validation attributes for edge validation
    /// </summary>
    public class CreateExampleRequest : BaseRequestDto
    {
        [Required(ErrorMessage = "Name is required")]
        [StringLength(100, ErrorMessage = "Name cannot exceed 100 characters")]
        public string Name { get; set; } = string.Empty;

        [StringLength(500, ErrorMessage = "Description cannot exceed 500 characters")]
        public string Description { get; set; } = string.Empty;
    }
}

File: infrastructures/ddd/typescript/src/api/dto/v1/requests/CreateExampleRequest.ts
import { BaseRequestDto } from '../infra/BaseRequestDto';

/**
 * Request DTO for creating a new example.
 * Request DTOs:
 * - Live in /api/dto/v1/requests/
 * - Suffix with "Request"
 * - Used for HTTP request payloads
 * - Transport only - NO business logic
 * - Validate at edge (API layer)
 * 
 * Example validation decorators (if using class-validator):
 * @IsNotEmpty()
 * @IsString()
 * @MaxLength(100)
 */
export class CreateExampleRequest extends BaseRequestDto {
  /**
   * The name of the example
   * Required, max 100 characters
   */
  public Name: string;

  /**
   * The description of the example
   * Optional
   */
  public Description: string;

  constructor() {
    super();
    this.Name = '';
    this.Description = '';
  }
}
